packages/react/src/components/Table/index.test.tsx

```diff
@@ -0,0 +1,314 @@
+  ];
+
+  tableItems.forEach((element) => {
+    expect(element).toHaveAttribute('data-foo', 'true');
+  });
+
+  expect(firstTableHeader).toHaveAttribute('scope', 'col');
+});
+
+test('should render the expected semantic HTML elements', () => {
+  renderDefaultTable();
+
+  const table = screen.getByRole('table');
+  const tableHead = screen.getByTestId('thead');
+  const tableBody = screen.getByTestId('tbody');
+  const tableRows = screen.getAllByRole('row');
+  const tableHeaders = screen.getAllByRole('columnheader');
+  const tableCells = screen.getAllByRole('cell');
+  const tableFooter = screen.getByTestId('tfoot');
+
+  const firstTableRow = tableRows[0];
+  const firstTableHeader = tableHeaders[0];
+  const firstTableCell = tableCells[0];
+
+  expect(table.tagName).toBe('TABLE');
+  expect(tableHead.tagName).toBe('THEAD');
+  expect(tableBody.tagName).toBe('TBODY');
+  expect(firstTableRow.tagName).toBe('TR');
+  expect(firstTableHeader.tagName).toBe('TH');
+  expect(firstTableCell.tagName).toBe('TD');
+  expect(tableFooter.tagName).toBe('TFOOT');
+});
+
+test('should render with border variant', () => {
+  render(
+    <Table variant="border">
+      <TableHead>
+        <TableRow>
+          <TableHeader>Header</TableHeader>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        <TableRow>
+          <TableCell>Cell</TableCell>
+        </TableRow>
+      </TableBody>
+    </Table>
+  );
+
+  expect(screen.getByRole('table')).toHaveClass('Table--border');
+});
+
+test('should render sort button and icons with sortDirection and onSort in Table', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader sortDirection={'none'} onSort={() => null}>
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByRole('button')).toBeInTheDocument();
+  expect(screen.getByRole('status').closest('.Icon--sort-triangle'));
+  expect(screen.getByRole('status')).toHaveTextContent('');
+});
+
+test('should render className "TableHeader--sorting" when actively sorting', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader sortDirection={'ascending'} onSort={() => null}>
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByRole('status').closest('.TableHeader--sort-ascending'));
+});
+
+test('should render triangle up Icon and ascending message when sortDirection is ascending', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader
+            sortDirection={'ascending'}
+            sortAscendingAnnouncement={'up and away'}
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByText('up and away')).toBeInTheDocument();
```

Comment:
Small suggestion to better validate that we're setting the announcement correctly.

```suggestion
  expect(screen.getByRole('status')).toHaveTextContent('up and away')
```